### DISCUSSION TITLE
Unsatisfied Link Error Ideas

### DISCUSSION THREAD

**Discussion by jkleiber** [team401/2024-Robot-Code#84] [Ideas]
Perhaps this will be useful in debugging the Utils.getCurrentTimeSeconds error: https://stackoverflow.com/questions/2446426/what-is-the-cause-of-an-unsatisfiedlinkerror

My suspicion is that we need to compare our build.gradle with the generated project's build.gradle (and maybe other build related scripts)

**Comment by honzikschenk** [comment]
﻿﻿﻿﻿﻿﻿﻿﻿ERROR ﻿﻿ 1 ﻿﻿ Unhandled exception: java.lang.UnsatisfiedLinkError: 'double com.ctre.phoenix6.Utils.getCurrentTimeSeconds()' ﻿﻿ com.ctre.phoenix6.Utils.getCurrentTimeSeconds(Native Method) ﻿﻿﻿
﻿﻿﻿﻿﻿﻿ Error at com.ctre.phoenix6.Utils.getCurrentTimeSeconds(Native Method): Unhandled exception: java.lang.UnsatisfiedLinkError: 'double com.ctre.phoenix6.Utils.getCurrentTimeSeconds()' ﻿
﻿﻿﻿﻿﻿﻿ 	at com.ctre.phoenix6.Utils.getCurrentTimeSeconds(Native Method) ﻿
﻿﻿﻿﻿﻿﻿ 	at frc.robot.Telemetry.<init>(Telemetry.java:66) ﻿
﻿﻿﻿﻿﻿﻿ 	at frc.robot.RobotContainer.<init>(RobotContainer.java:51) ﻿
﻿﻿﻿﻿﻿﻿ 	at frc.robot.Robot.robotInit(Robot.java:61) ﻿
﻿﻿﻿﻿﻿﻿ 	at org.littletonrobotics.junction.LoggedRobot.startCompetition(LoggedRobot.java:79) ﻿
﻿﻿﻿﻿﻿﻿ 	at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:365) ﻿
﻿﻿﻿﻿﻿﻿ 	at edu.wpi.first.wpilibj.RobotBase.startRobot(RobotBase.java:453) ﻿
﻿﻿﻿﻿﻿﻿ 	at frc.robot.Main.main(Main.java:23) ﻿
﻿﻿﻿﻿﻿﻿  ﻿
﻿﻿﻿﻿﻿﻿﻿﻿Warning ﻿﻿ 1 ﻿﻿ The robot program quit unexpectedly. This is usually due to a code error.
  The above stacktrace can help determine where the error occurred.
  See https://wpilib.org/stacktrace for more information. ﻿﻿ edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:379) ﻿﻿﻿
﻿﻿﻿﻿﻿﻿ ﻿Warning﻿ at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:379): The robot program quit unexpectedly. This is usually due to a code error. ﻿
﻿﻿﻿﻿﻿﻿   The above stacktrace can help determine where the error occurred. ﻿
﻿﻿﻿﻿﻿﻿   See https://wpilib.org/stacktrace for more information. ﻿
﻿﻿﻿﻿﻿﻿﻿﻿ERROR ﻿﻿ 1 ﻿﻿ The startCompetition() method (or methods called by it) should have handled the exception above. ﻿﻿ edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:386) ﻿﻿﻿
﻿﻿﻿﻿﻿﻿ Error at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:386): The startCompetition() method (or methods called by it) should have handled the exception above.

**Reply by jkleiber** [reply]
Can you put up a draft PR for the code that has the link error? Also maybe a link to the temporary project that does work would be helpful

I think this isn't related to the change itself and it would be good to confirm we can't reproduce this in sim. If we can't reproduce in sim, then this is likely due to something related to the deploy build step

**Reply by honzikschenk** [reply]
Sure

**Reply by honzikschenk** [reply]
The error happens in the sim as well. I will investigate further into what the new tuner constants changed.

**Reply by honzikschenk** [reply]
@jkleiber Just found the issue. It was with the way we constructed `CommandSwerveDrivetrain drivetrain;` It should now be fixed in #87

**Comment by honzikschenk** [comment]
Let's keep this open until we confirm that the issue is fixed on the real robot.

**Comment by honzikschenk** [comment]
Fixed. DriveTelemetry was grabbing an unconstructed drivetrain causing the code to crash.
